content/articles/building-a-routing-service-using-mapquest-api/index.md

```diff
@@ -0,0 +1,407 @@
+
+        #form {
+            padding: 0;
+            margin: 0;
+        }
+
+        input:nth-child(1) {
+            margin-bottom: 10px;
+        }
+
+        #button {
+            display: none;
+        }
+
+        .leaflet-top .leaflet-control {
+            margin-top: 180px;
+        }
+    </style>
+</head>
+
+<body>
+    <div id="map"></div>
+    <div class="formBlock">
+        <form id="form">
+            <input type="text" name="start" class="input" id="start" placeholder="Choose starting point" />
+            <input type="text" name="end" class="input" id="destination" placeholder="Choose ending point" />
+            <button id="button" type="submit">Go</button>
+        </form>
+    </div>
+</body>
+<script>
+    // default map layer
+    let map = L.map('map', {
+        layers: MQ.mapLayer(),
+        center: [9.0820, 8.6753],
+        zoom: 8
+    });
+    function runDirection(start, end) {
+
+        // recreating new map layer after removal
+        map = L.map('map', {
+            layers: MQ.mapLayer(),
+            center: [9.0820, 8.6753],
+            zoom: 8
+        });
+
+        var direction = MQ.routing.directions();
+
+        direction.route({
+            locations: [
+                start,
+                end
+            ]
+        });
+
+
+        MainRouteLayer = MQ.Routing.RouteLayer.extend({
+            initstartmarker: (location) => {
+                var icon1;
+                var startmarker;
+
+                icon1 = L.icon({
+                    iconUrl: 'red.png',
+                    iconSize: [20, 29],
+                    iconAnchor: [10, 29],
+                    popupAnchor: [0, -29]
+                });
+
+                startmarker = L.marker(location.latLng, { icon: icon1 }).addTo(map);
+
+                return startmarker;
+            },
+
+            initEndmarker: (location) => {
+                var icon2;
+                var endmarker;
+
+                icon2 = L.icon({
+                    iconUrl: 'blue.png',
+                    iconSize: [20, 29],
+                    iconAnchor: [10, 29],
+                    popupAnchor: [0, -29]
+                });
+
+                endmarker = L.marker(location.latLng, { icon: icon2 }).addTo(map);
+
+                return endmarker;
+            }
+        });
+
+        map.addLayer(new MainRouteLayer({
+            directions: direction,
+            fitBounds: true
+        }));
+    }
+
+    function submitForm(event) {
+        event.preventDefault();
+
+        map.remove();
+
+        start = document.getElementById("start").value;
+        end = document.getElementById("destination").value;
+
+        runDirection(start, end);
+
+        document.getElementById("form").reset();
+    }
+    const form = document.getElementById('form');
+
+    form.addEventListener('submit', submitForm);
+</script>
+
+</html>
+```
+![final-result](engineering-education/building-a-routing-service-using-mapquest-api/final.gif)
+
```

Comment:
From this gif image, I've noticed you're running it on the browser. You can explain to the reader in a paragraph or two on how to do this.